x-pack/plugins/case/common/api/cases/comment.ts

```diff
@@ -90,3 +121,10 @@ export type CommentPatchRequest = rt.TypeOf<typeof CommentPatchRequestRt>;
 export type CommentPatchAttributes = rt.TypeOf<typeof CommentPatchAttributesRt>;
 export type CommentRequestUserType = rt.TypeOf<typeof ContextTypeUserRt>;
 export type CommentRequestAlertType = rt.TypeOf<typeof ContextTypeAlertRt>;
+
+/**
+ * This type represents a generated alert (or alerts) from a rule. The difference between this and a user alert
+ * is that this format expects the included alerts to have the structure { _id: string }. When it is saved the outer
+ * object will be stripped off and the _id will be stored in the alertId field.
+ */
+export type CommentRequestGeneratedAlertType = rt.TypeOf<typeof ContextTypeGeneratedAlertRt>;
```

Comment:
I found it a bit confusing that `ContextTypeAlertRt` can have as a type `CommentType.generatedAlert` and `CommentRequestGeneratedAlertType` can have as a type only `CommentType.generatedAlert`. If this type is used only by the case connnector then maybe `CommentRequestGeneratedByRuleType` make more sense. If missing something please ignore :)